lib/cereal/postgres/postgres.go

```diff
@@ -566,6 +561,29 @@ func (pg *PostgresBackend) DequeueWorkflow(ctx context.Context, workflowNames []
 	return event, workc, nil
 }
 
+func getTaskResult(ctx context.Context, tx *sql.Tx, taskResultID sql.NullInt64) (*backend.TaskResult, error) {
+	if !taskResultID.Valid {
+		return nil, errors.New("invalid task result id for completed task event")
+	}
+
+	row := tx.QueryRowContext(ctx, getTaskResultQuery, taskResultID.Int64)
+	tr := backend.TaskResult{}
+	err := row.Scan(
+		&tr.TaskName, &tr.Parameters, &tr.Status, &tr.ErrorText, &tr.Result)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// This is unlikely to be a transient problem
+			// that aborting and retrying will fix. Thus,
+			// we move on and allow the workflow to handle
+			// the nil result.
+			return nil, nil
```

Comment:
instead of nil, can we return a task result that returns an error

Reply:
Oh, I like that idea.  Since then, if the user didn't need the task result, fine, but if they did they will definitely get an error when asking for it and don't have to do some nil checks on whatever they dereferenced.

Reply:
Updated this to return an error, I thought about making it a completely different type, but I think this works for now.